students/AndyKwok/lesson06/test_mailroom.py

```diff
@@ -0,0 +1,69 @@
+# Description: Mailroom Program Tester
+# Author: Andy Kwok
+# Last Updated: 08/11/2018
+# ChangeLog: Initializing
+
+#!/usr/bin/env python3
+
+import unittest
+import io
+import os
+import sys
+import mailroom_part4 as atest
+from unittest.mock import patch
+
+class test_mailroom_part4(unittest.TestCase):
+    def setUp(self):
+        self.donor = 'Donor I'
+        self.donation = [300.00]
+        self.donor2 = 'Donor B'
+        self.donation2 = [1.00]           
+        self.filelist = ['Donor_A.txt', 'Donor_B.txt', 'Donor_C.txt']
+    
+    def test_report_select(self):
+        expected = {'Donor C': [14349.00, 175.95], 'Donor A': [2300.30, 1000.01, 2000.00], 'Donor B': [1394.12, 1.0], 'Donor I': [300.00]}
+        atest.report_select()
+        actual = atest.database
+        self.assertEqual(expected, actual)
```

Comment:
this function is not testing `report_select` function; in order to test it you need to modify the function to return a list or a string representing report rows - there would be a separate function to actually print your report.